packages/cli/src/commands/set-version.ts

```diff
@@ -0,0 +1,169 @@
+import chalk from "chalk";
+import { existsSync, readFileSync, rmSync, writeFileSync } from "fs";
+import path from "path";
+import type { CommandModule } from "yargs";
+import { logError, MUDError } from "../utils/errors.js";
+
+type Options = {
+  backup?: boolean;
+  force?: boolean;
+  restore?: boolean;
+  mudVersion?: string;
+};
+
+const BACKUP_FILE = ".mudbackup";
+
+const commandModule: CommandModule<Options, Options> = {
+  command: "set-version",
+
+  describe: "Install a custom MUD version and optionally backup the previously installed version",
+
+  builder(yargs) {
+    return yargs.options({
+      backup: { type: "boolean", description: "Back up the current MUD versions to `.mudinstall`" },
+      force: {
+        type: "boolean",
+        description: "Backup fails if a .mudinstall file is found, unless --force is provided",
+      },
+      restore: { type: "boolean", description: "Restore the previous MUD versions from `.mudinstall`" },
+      mudVersion: { alias: "v", type: "string", description: "The MUD version to install" },
+    });
+  },
+
+  async handler(options) {
+    try {
+      if (!options.mudVersion && !options.restore) {
+        throw new MUDError(`Version parameter is required unless --restore is provided.`);
+      }
+
+      // Resolve the `canary` version number if needed
+      options.mudVersion =
+        options.mudVersion === "canary" ? await getCanaryVersion("@latticexyz/world") : options.mudVersion;
+
+      // Read the current package.json
+      const rootPath = "./package.json";
+      const { workspaces } = updatePackageJson(rootPath, options);
+
+      // Load the package.json of each workspace
+      if (workspaces) {
+        for (const workspace of workspaces) {
+          const filePath = path.join(workspace, "/package.json");
+          updatePackageJson(filePath, options);
+        }
+      }
+    } catch (e) {
+      logError(e);
+    } finally {
+      process.exit(0);
+    }
+  },
+};
+
+function updatePackageJson(filePath: string, options: Options): { workspaces?: string[] } {
+  const { backup, restore, force, mudVersion } = options;
+  const backupFilePath = path.join(path.dirname(filePath), BACKUP_FILE);
+
+  // If `backup` is true and force not set, check if a backup file already exists and throw an error if it does
+  if (backup && !force && existsSync(backupFilePath)) {
+    throw new MUDError(
+      `A backup file already exists at ${backupFilePath}.\nUse --force to overwrite it or --restore to restore it.`
+    );
+  }
+
+  const packageJson = readPackageJson(filePath);
+
+  // Load .mudinstall if `restore` is true
+  const backupJson = restore ? readPackageJson(backupFilePath) : undefined;
+
+  // Find all @latticexyz dependencies
+  const mudDependencies: Record<string, string> = {};
+  for (const key in packageJson.dependencies) {
+    if (key.startsWith("@latticexyz")) {
```

Comment:
kinda feel like this is worth lifting into a variable/constant, to make it easier to change if we decide later to move to e.g. `@muddev`